From 21c6b06fcf1560e1751db187dcc717c137de6547 Mon Sep 17 00:00:00 2001
From: Kentaro Hayashi <hayashi@clear-code.com>
Date: Mon, 22 Jan 2018 16:30:05 +0900
Subject: [PATCH] nginx: fix build error on rawhide

Since crypt library is switched into libxcrypt on Fedora 28 (instead
of glibc with libcrypt), it turned out that FTBFS occurs. It seems
that workaround code causes it. The bug itself was already fixed in
glibc 2.3.2, so there is no need to use that workaround anymore.

Signed-off-by: Kentaro Hayashi <hayashi@clear-code.com>
---
 vendor/nginx-1.13.8/src/os/unix/ngx_user.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/vendor/nginx-1.14.0/src/os/unix/ngx_user.c b/vendor/nginx-1.14.0/src/os/unix/ngx_user.c
index 7ebe2b576..23a070147 100644
--- a/vendor/nginx-1.14.0/src/os/unix/ngx_user.c
+++ b/vendor/nginx-1.14.0/src/os/unix/ngx_user.c
@@ -21,8 +21,9 @@ ngx_libc_crypt(ngx_pool_t *pool, u_char *key, u_char *salt, u_char **encrypted)
     struct crypt_data   cd;
 
     cd.initialized = 0;
-#ifdef __GLIBC__
-    /* work around the glibc bug */
+#if __GLIBC__ == 2 && __GLIBC_MINOR__ <= 3
+    /* work around the glibc bug, it was fixed in 2.3.2.
+       see https://www.ruby-forum.com/topic/4412553 */
     cd.current_salt[0] = ~salt[0];
 #endif
 
-- 
2.15.1

